endpoints/openrtb2/amp_auction.go

```diff
@@ -168,10 +166,13 @@ func (deps *endpointDeps) AmpAuction(w http.ResponseWriter, r *http.Request, _ h
 	} else {
 		labels.CookieFlag = metrics.CookieFlagNo
 	}
-	labels.PubID = getAccountID(req.Site.Publisher)
+	labels.PubID = getAccountID(reqWrapper.Site.Publisher)
 	// Look up account now that we have resolved the pubID value
 	account, acctIDErrs := accountService.GetAccount(ctx, deps.cfg, deps.accounts, labels.PubID)
 	if len(acctIDErrs) > 0 {
+		reqWrapper.RebuildRequest()
```

Comment:
Yes. Normally this will be synced later, but in this case we have a fatal error and need to rebuild the request before passing it to the analytics object.